src/cpl/share_esmf/IrrigationStreamMod.F90

```diff
@@ -0,0 +1,520 @@
+  type(shr_strdata_type)  :: sdat_irrig                 ! irrigation input data stream
+  
+  integer :: ism                                        ! irrigation stream index
+  integer     , parameter :: numFields = 8              ! number of fields to build field string
+  character(cl)           :: stream_varnames(numFields) ! field string
+  character(len=CL)       :: stream_lev_dimname = 'cft' ! name of vertical layer dimension
+  integer :: num_irrig_cft                              ! number of irrigated cfts on stream file
+
+  integer, allocatable    :: g_to_ig(:)                    ! Array matching gridcell index to data index
+  logical                 :: irrig_ignore_data_if_missing  ! If should ignore overridding a point with soil moisture data
+                                                           ! from the streams file, if the streams file shows that point 
+                                                           ! as missing (namelist item)
+  ! !PRIVATE TYPES:
+  character(len=*), parameter, private :: sourcefile = &
+       __FILE__
+  character(*), parameter :: u_FILE_u = &
+       __FILE__
+
+!==============================================================================
+contains
+!==============================================================================
+  
+  subroutine PrescribedIrrigationInit(bounds)
+    !
+    ! Initialize data stream information for soil moisture.
+    !
+    ! !USES:
+    ! !USES: UNSURE
+    use clm_varctl       , only : inst_name
+    use clm_time_manager , only : get_calendar
+    use ncdio_pio        , only : pio_subsystem
+    use shr_pio_mod      , only : shr_pio_getiotype
+    use clm_nlUtilsMod   , only : find_nlgroup_name
+    
+    use shr_stream_mod   , only : shr_stream_file_null
+    use shr_string_mod   , only : shr_string_listCreateField
+    use clm_varpar       , only : nlevsoi
+    ! !ARGUMENTS:
+    implicit none
+    type(bounds_type), intent(in)  :: bounds          ! bounds
+    !
+    ! !LOCAL VARIABLES:
+    integer            :: rc                         ! error coe
+    integer            :: i                          ! index
+    integer            :: stream_year_first_irrig    ! first year in Ustar stream to use
+    integer            :: stream_year_last_irrig     ! last year in Ustar stream to use
+    integer            :: model_year_align_irrig     ! align stream_year_first_soilm with 
+    integer            :: nu_nml                     ! unit for namelist file
+    integer            :: nml_error                  ! namelist i/o error flag
+    integer            :: irrig_offset               ! Offset in time for dataset (sec)
+    character(len=CL)  :: stream_fldfilename_irrig   ! ustar stream filename to read
+    character(len=CL)  :: irrig_tintalgo = 'nearest'  ! Time interpolation alogrithm
+    character(len=CL)  :: stream_mapalgo = 'nn'
+    real(r8)           :: stream_dtlimit = 15._r8
+    character(len=CL)  :: stream_taxmode = 'cycle'
```

Comment:
- [ ] It would probably make more sense for the irrigation stream file's `stream_taxmode` to be `extend`. That way, with an input file that covers (e.g.) 1980–2014, 1975 would use the 1980 value instead of the 2010 value.